src/vstart.sh

```diff
@@ -992,7 +992,7 @@ EOF
         ceph_adm config-key set mgr/cephadm/ssh_identity_pub -i ~/.ssh/id_rsa.pub
         ceph_adm mgr module enable cephadm
         ceph_adm orch set backend cephadm
-        ceph_adm orch host add $HOSTNAME
+        ceph_adm orch host add $(hostname --fqdn)
```

Comment:
don't know if we should do that this way. We have to make sure, the first parameter of `host add` matches what `ceph * metadata` returns as hostnames. 

If `metadata` also returns the fqdn, this patch is good. If `metadata` returns the short name, I think:

```suggestion
        ceph_adm orch host add $HOSTNAME $(hostname --fqdn)
```

would be better.

Reply:
the result from the `hostname` command must also match what is given here or `bin/cephadm` will complain:

```
$ ceph orch host add host1.site 192.168.0.1
Error ENOENT: New host host1.site (192.168.0.1) failed check: ['INFO:cephadm:podman|docker (/usr/bin/podman) is present', 'INFO:cephadm:systemctl is present', 'INFO:cephadm:lvcreate is present', 'INFO:cephadm:Unit chronyd.service is enabled and running', 'INFO:cephadm:Hostname "host1.site" matches what is expected.', 'ERROR: hostname "host1" does not match expected hostname "host1.site"']
```

Reply:
I thought the second parameter to `orch host add` is an IP address?

Reply:
> I thought the second parameter to `orch host add` is an IP address?

it can be any resolvable address (hostname or IP), but I don't think simply adding the second param will solve this error ...

Reply:
> the result from the `hostname` command must also match what is given here or `bin/cephadm` will complain

You're right. `vstart` does the wrong thing here. 

```suggestion
        ceph_adm orch host add $(hostname)
```

should be better. Neither `hostname -s` nor `hostname --fqdn` is correct.